apps/server/src/shared/infra/s3-client/s3-client.module.ts

```diff
@@ -1,44 +1,41 @@
 import { S3Client } from '@aws-sdk/client-s3';
 import { DynamicModule, Module } from '@nestjs/common';
-import { LoggerModule } from '@src/core/logger';
-import { S3_CLIENT, S3_CONFIG } from './constants';
+import { LegacyLogger, LoggerModule } from '@src/core/logger';
 import { S3Config } from './interface';
 import { S3ClientAdapter } from './s3-client.adapter';
 
-@Module({
-	imports: [LoggerModule],
-	providers: [S3ClientAdapter],
-})
+const createS3ClientAdapter = (config: S3Config, legacyLogger: LegacyLogger) => {
+	const { region, accessKeyId, secretAccessKey, endpoint } = config;
+
+	const s3Client = new S3Client({
+		region,
+		credentials: {
+			accessKeyId,
+			secretAccessKey,
+		},
+		endpoint,
+		forcePathStyle: true,
+		tls: true,
+	});
+	return new S3ClientAdapter(s3Client, config, legacyLogger);
+};
+
+@Module({})
 export class S3ClientModule {
-	static register(options: S3Config): DynamicModule {
-		const providers = [
-			{
-				provide: S3_CONFIG,
-				useValue: options,
-			},
+	static register(options: S3Config[]): DynamicModule {
+		const providers = options.flatMap((configs) => [
```

Comment:
```suggestion
	static register(configs: S3Config[]): DynamicModule {
		const providers = configs.flatMap((config) => [
```
Du gehst über einen Array von _configs_ und mappst dann jeweils eine _config_, oder?

Reply:
Jap